nemo_curator/stages/text/embedders/base.py

```diff
@@ -62,33 +60,23 @@ def outputs(self) -> tuple[list[str], list[str]]:
     def setup(self, _: WorkerMetadata | None = None) -> None:
         """Load the model for inference."""
         self.model = AutoModel.from_pretrained(self.model_identifier, local_files_only=True)
-        self.model.eval()
-        self.model.to("cuda")
+        self.model.eval().to("cuda")
 
     def process_model_output(
         self, outputs: torch.Tensor, model_input_batch: dict[str, torch.Tensor] | None = None
     ) -> torch.Tensor:
         """Process model outputs to create embeddings."""
         if self.pooling == "mean_pooling":
-            return self._mean_pooling(outputs, model_input_batch[ATTENTION_MASK_COLUMN])
+            return self._mean_pooling(outputs, model_input_batch[ATTENTION_MASK_COLUMN]).cpu()
         else:
-            return self._get_last_token(outputs, model_input_batch[ATTENTION_MASK_COLUMN])
+            return self._get_last_token(outputs, model_input_batch[ATTENTION_MASK_COLUMN]).cpu()
 
-    def collect_outputs(self, processed_outputs: list[torch.Tensor]) -> cp.ndarray:
-        """Collect embeddings into a cupy array."""
-        # TODO : benchmarking this and maybe stay in cpu land
-        cupy_array_embeddings = [cp.asarray(emb_chunk) for emb_chunk in processed_outputs]
-        return cp.concatenate(cupy_array_embeddings, axis=0)
+    def collect_outputs(self, processed_outputs: list[torch.Tensor]) -> list[list[float]]:
+        return torch.cat(processed_outputs, dim=0).numpy().tolist()
```

Comment:
Is there a requirement here of this being float32/float16 for the downstream models.  With autocast this might become `float16/bfloat16`  etc and unsure if that is a problem for the other downstream models ? 


```suggestion
        return torch.cat(processed_outputs, dim=0).cpu().to(torch.float32).numpy().tolist()
```